Error when building demo project with AOT

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION YOUR ISSUE MIGHT BE CLOSED WITHOUT INVESTIGATING
-->
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### OS and Version?
<!--
> Windows 7, 8 or 10. Linux (which distribution).macOS(Yosemite ? El Capitan? Sierra ?)
-->
* Mac os Sierra (10.12.6)
* Windows 7

### Node / Npm or Yarn Version
<!--
Output from: `node --version`, 
Output from: `yo --version`, 
Output from: `npm -v`, or 
Output from: `yarn --version` , if you are using Yarn instead
-->
```shell
» node --version
v7.7.1
```
```shell
» yo --version
2.0.0
```
```shell
» yarn --version
0.22.0
```

### Targeted Angular version
<!-- 2.x.x, 4.x.x etc. -->
Tested with both 2.x.x and 4.x.x

### Repro steps.
<!--
Simple steps to reproduce this bug.
Please include: commands run, packages added, related code changes.
A link to a sample repo would help too.
-->
```shell
» yo ngx-library

     _-----_     ╭──────────────────────────╮
    |       |    │  Welcome to the wondrous │
    |--(o)--|    │     ngx-library@3.1.1    │
   `---------´   │        generator!        │
    ( _´U`_ )    ╰──────────────────────────╯
    /___A___\   /
     |  ~  |     
   __'.___.'__   
 ´   `  |° ´ Y ` 

? (1/12) What is your name? You name
? (2/12) What is your email address? email@email.fr
? (3/12) What is your Github username? username
? (4/12) What is the name of your project? ngx-lib
? (5/12) What is the version of your project? 0.0.1
? (6/12) What is the description of your project? Angular library built with ❤ u
sing ngx-library yeoman generator.
? (7/12) What keywords best describe your project (comma-separated)? ng,angular,
library
? (8/12) What minimal version of Angular do you want to base your library upon? 
4.X.X
? (9/12) What prefix would you like to use to name your components, directives,.
..? ngx-lib
? (10/12) What Angular modules will your library use? core, common, http, compil
er, platform-browser-dynamic, platform-browser
? (11/12) Do You want to use Greenkeepeer (to automatically keep your dependenci
es up-to-date)? No
? (12/12) Do You want to use Compodoc (to generate your Angular project document
ation)? No

[...]

» gulp build:demo
```

### The log given by the failure.
<!-- Normally this include a stack trace and some more information. -->
```shell
[17:39:29] Using gulpfile /private/tmp/ngx-test/gulpfile.js
[17:39:29] Starting 'build:demo'...
[17:39:29] No 'node_modules' found in 'demo/'. Installing dependencies for you..
yarn install v0.22.0
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.84s.
yarn install v0.22.0
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
$ npm link ngx-lib
/private/tmp/ngx-test/demo/node_modules/ngx-lib -> /usr/local/lib/node_modules/ngx-lib -> /private/tmp/ngx-test/dist
Done in 288.09s.
Date: 2017-08-28T15:44:42.052Z                                                y Hash: ef70c651342119cfd414
Time: 19218ms
chunk {0} polyfills.b59a8422e26e64d12be4.bundle.js (polyfills) 63.9 kB {3} [initial] [rendered]
chunk {1} main.f4fab8807a794a7599a3.bundle.js (main) 61.7 kB {3} [initial] [rendered]
chunk {2} styles.8d8889034cdf8cb01f46.bundle.css (styles) 166 kB {3} [initial] [rendered]
chunk {3} inline.f80d2f9e96d89d576d7c.bundle.js (inline) 1.45 kB [entry] [rendered]

ERROR in Error encountered resolving symbol values statically. Could not resolve ngx-lib relative to /private/tmp/ngx-test/demo/src/app/app.module.ts., resolving symbol AppModule in /private/tmp/ngx-test/demo/src/app/app.module.ts, resolving symbol AppModule in /private/tmp/ngx-test/demo/src/app/app.module.ts, resolving symbol AppModule in /private/tmp/ngx-test/demo/src/app/app.module.ts
ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/private/tmp/ngx-test/demo/src'
resolve './$$_gendir/app/app.module.ngfactory' in '/private/tmp/ngx-test/demo/src'
  using description file: /private/tmp/ngx-test/demo/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /private/tmp/ngx-test/demo/package.json (relative path: ./src)
    using description file: /private/tmp/ngx-test/demo/package.json (relative path: ./src/$$_gendir/app/app.module.ngfactory)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        /private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory.js doesn't exist
      as directory
        /private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory doesn't exist
[/private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory]
[/private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory.ts]
[/private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory.js]
[/private/tmp/ngx-test/demo/src/$$_gendir/app/app.module.ngfactory]
 @ ./src/main.ts 3:0-74
 @ multi ./src/main.ts
[17:44:42] ng command failed. See below for errors.
```

### Desired functionality.
<!--
What would like to see implemented?
What is the usecase?
-->
Be able to use library within project using AOT

### Mention any other details that might be useful.
<!-- Please include a link to the repo if this is related to an OSS project. -->
I have tested with both target version 2.X and 4.X and I have tested under Windows and Macos and I have the same error each time.

--- a/app/templates/_gulpfile.js
+++ b/app/templates/_gulpfile.js
@@ -306,7 +306,9 @@ gulp.task('npm-package', (cb) => {
   // copy the needed additional files in the 'dist' folder
   pump(
     [
-      gulp.src(['README.md', 'LICENSE', 'CHANGELOG.md',
+      gulp.src(['README.md', 'LICENSE', 'CHANGELOG.md',<% if(ngVersion === '2.0.0'){ %>
+      `${config.buildDir}/lib-es5/**/*.js`,
+      `${config.buildDir}/lib-es5/**/*.js.map`,<% } %>
       `${config.buildDir}/lib-es5/**/*.d.ts`,
       `${config.buildDir}/lib-es5/**/*.metadata.json`]),
       gulpFile('package.json', JSON.stringify(targetPkgJson, null, 2)),

